diff -Naur centericq-4.21.0/src/icqface.cc centericq-4.21.0.mod/src/icqface.cc
--- centericq-4.21.0/src/icqface.cc	2005-08-28 08:33:21.000000000 +0700
+++ centericq-4.21.0.mod/src/icqface.cc	2006-12-08 18:28:13.000000000 +0600
@@ -656,6 +656,7 @@
 	if(c->getstatus() == offline)
 	if(conf.gethideoffline())
 	if(!c->hasevents())
+	if(c->getdispnick()[0] != '~')
 	if(c->inlist() || !h.logged() || h.getCapabs().count(hookcapab::cltemporary)) {
 	    continue;
 	}
